x-pack/plugins/cases/server/connectors/cases/utils.ts

```diff
@@ -52,9 +52,33 @@ export const convertValueToString = (value: unknown): string => {
   return toString(value);
 };
 
-export const buildRequiredCustomFieldsForRequest = (
-  customFieldsConfiguration?: CustomFieldsConfiguration
+export const buildCustomFieldsForRequest = (
+  customFieldsConfiguration?: CustomFieldsConfiguration,
+  templateCustomFields?: CaseCustomFields
 ): CaseRequestCustomFields => {
+  if (templateCustomFields && templateCustomFields.length) {
+    return templateCustomFields.map((templateCustomField) => {
```

Comment:
Is it possible for the template to not have some custom fields? If yes, maybe we can iterate through the configuration as we do below and try to find the custom field in the template.

Reply:
we add newly added custom field to template or remove deleted custom field from template when updating the configuration. So I think templates will always be up to date with all existing custom fields, wdyt?

Reply:
I tried by using only the API and I created a template without custom fields even if I have some custom fields configured. I think is better to iterate through the configuration just to be sure. Maybe we can combine with the code below that iterates through the configuration and do it in one loop.

Reply:
[Updated](https://github.com/elastic/kibana/pull/190701/commits/4e3e22c014d13c0209797262adb3a4bcac42ec51#diff-23ba27ef6d5bce05f752c6e3e8a087aa07190978998d6ef8549763896e7ff3e5). Didn't create single loop as it made code more confusing 😅